### PR TITLE
Prepare for ruby 3.0

### PR DESCRIPTION
**Pull request by tak1n**
Ruby 3.0 is going to be released on 25th of December, therefore preparing for it sounds like a good idea to me.

Also another question, it is a common best practice to not check in `Gemfile.lock` to version control for rubygems, should I add it to `.gitignore`?

### CODE REVIEW

File: fast_excel.gemspec [hunk 1]
```diff
@@ -7,7 +7,7 @@ Gem::Specification.new do |s|
   s.summary     = %q{Ultra Fast Excel Writer}
   s.description = "Wrapper for libxlsxwriter using ffi"
   s.license     = 'MIT'
-  s.required_ruby_version = '~> 2.0'
+  s.required_ruby_version = '~> 3.0'
```
Comment (dlozano) [inline fast_excel.gemspec:10]: Why is this needed?!
It makes sense adding 3.0 to Travis matrix, but why do you need so break compatibility?

**Review by dlozano — CHANGES_REQUESTED** [review]

File: fast_excel.gemspec (hunk 1 above)
Comment (tak1n) [reply fast_excel.gemspec:10]: Sry my fault, it should be `>= 2.0`, so we allow ruby 3 to be used but require at least 2.0.

Current `~> 2.0` will only allow 2.x but not 3.

**Review by dlozano — APPROVED** [review]

File: fast_excel.gemspec (hunk 1 above)
Comment (Paxa) [reply fast_excel.gemspec]: I don't think anyone still use ruby 1.9 now, but if there will be ruby 4.0 or ruby 5.0 with breaking changes I don't want bundler to choose older version because because we have `>= 2.0`. And probably we can't set it to `['>= 2.0', '< 4.0']`

So probably `< 4.0` will be more safe

**Comment by Paxa** [comment]
Thanks for PR, I will try to fix tests later and merge. About `Gemfile.lock` not sure if it affect tests but I use it to track dependency versions that I use in library development. Also I can help others to have same dependency versions if they experience any problems

File: fast_excel.gemspec (hunk 1 above)
Comment (tak1n) [reply fast_excel.gemspec]: According to https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Gemspec/RequiredRubyVersion it can be set to 

```
spec.required_ruby_version = ['>= 2..0', '< 4.0']
```

Will change that.

**Comment by tak1n** [comment]
> Thanks for PR, I will try to fix tests later and merge. About `Gemfile.lock` not sure if it affect tests but I use it to track dependency versions that I use in library development. Also I can help others to have same dependency versions if they experience any problems

Fine for me, I'm just used to not checking it in on various gems I worked on - https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/ :v:
